connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -62,21 +77,69 @@ public static ConnectLocation getInstance() {
         return INSTANCE;
     }
 
+    /**
+     * This is a required method to provide the Location SDK with a listener for state change events from {@link ConnectButton}.
+     * The Connect Location SDK will not be able to handle state change events and register/unregister geofences if you don't call this method.
+     * @param connectButton button instance initialized to display the connection.
+     */
     public void setUpWithConnectButton(ConnectButton connectButton) {
         connectButton.addButtonStateChangeListener(this);
     }
 
+    /**
+     * Use this method to add a listener to be notified when the connection is enabled and location permissions need to be granted.
+     * Calling this method is required to make sure the necessary permissions are granted when the connection is enabled.
+     * If your app is using the ACCESS_FINE_LOCATION permission for other use cases and can guarantee that the permission is always granted,
+     * or has checks to make sure permission is requested when required, you can consider this method optional.
+     * @param permissionCallback {@link LocationPermissionCallback to be registered}
+     */
+    public void setUpPermissionPromptCallback(LocationPermissionCallback permissionCallback) {
+        this.permissionCallback = permissionCallback;
+    }
+
+    /**
+     * Use this method to need to check whether location permission is required for the connection,
+     * based on the latest status of the connection as well as that of its location feature.
+     * This is a one time permission check function unlike {@link ConnectLocation#setUpPermissionPromptCallback(LocationPermissionCallback)}
+     * You can call this method from places where you are managing other user permissions or in your home activity,
+     * but only after first calling {@link ConnectLocation#init(Context, String, ConnectionApiClient)} or {@link ConnectLocation#init(Context, String, UserTokenProvider)}
+     * @param activity the activity from where you call this method
+     * @param lifecycle lifecycle of the activity
+     * @param connectionId connection Id of the connection for which permission needs to be checked.
+     * @param permissionCallback {@link LocationPermissionCallback callback to invoke when location permission is missing.}
+     */
+    public void checkPermission(Activity activity, Lifecycle lifecycle, String connectionId, LocationPermissionCallback permissionCallback) {
```

Comment:
I'm not sure if we should dictate how the developers handle lifecycle and the API call in this method: we don't know if the developers actually use `Lifecycle` to manage it. 

Instead of trying to solve the problem ourselves, we can let this method return `PendingResult<Connection>`, which can be used to cancel the call.